crates/core/src/task_center.rs

```diff
@@ -805,15 +805,21 @@ impl TaskCenterInner {
         }
         self.cancel_tasks(Some(TaskKind::ClusterController), None)
             .await;
+        // stop accepting ingress
+        self.cancel_tasks(Some(TaskKind::IngressServer), None).await;
+        // PPM will shutdown running processors
+        self.cancel_tasks(Some(TaskKind::PartitionProcessorManager), None)
+            .await;
+        self.initiate_managed_runtimes_shutdown();
+        // global shutdown trigger
+        self.cancel_tasks(None, None).await;
         tokio::join!(
             self.cancel_tasks(Some(TaskKind::RpcServer), None),
             self.cancel_tasks(Some(TaskKind::ConnectionReactor), None),
             // note that this one will shutdown after connections are terminated.
             // It will not be aborted on the shutdown signal itself.
             self.cancel_tasks(Some(TaskKind::SocketHandler), None)
         );
```

Comment:
Wouldn't `self.cancel_tasks(None, None).await` already cancel all tasks so that the `tokio::join!()` block is no longer needed?

Reply:
Yes. I'll remove it, but this part is still not the best. Expect this to change a again soon.